phalcon/cache/backend/file.zep

```diff
@@ -160,6 +173,11 @@ class File extends Backend implements BackendInterface
 					let ret = frontend->afterRetrieve(cachedContent);
 					return ret;
 				}
+			} else {
+				/**
+				 * As the content is expired, deleting the key
+				 */
+				this->delete(keyName);
```

Comment:
Yes, if the cache is valid, it returns its value. Check out the current code in master: https://github.com/phalcon/cphalcon/blob/master/phalcon/cache/backend/file.zep#L163 

`if !(time() - ttl > modifiedTime) {` e.g. if the cache is not expired, return its value. There is no else clause, nor return after the if. I've just appended an else clause, where we delete the cache file from the filesystem as it is expired, and still not returning anything as we don't at the moment.